src/DetailsView/components/load-assessment-button.tsx

```diff
@@ -3,31 +3,104 @@
 
 import { AssessmentDataParser } from 'common/assessment-data-parser';
 import { InsightsCommandButton } from 'common/components/controls/insights-command-button';
+import { Tab } from 'common/itab';
+import { AssessmentStoreData } from 'common/types/store-data/assessment-result-data';
+import { TabStoreData } from 'common/types/store-data/tab-store-data';
+import { VersionedAssessmentData } from 'common/types/versioned-assessment-data';
+import { UrlParser } from 'common/url-parser';
 import { DetailsViewActionMessageCreator } from 'DetailsView/actions/details-view-action-message-creator';
+import { LoadAssessmentDialog } from 'DetailsView/components/load-assessment-dialog';
 import { LoadAssessmentHelper } from 'DetailsView/components/load-assessment-helper';
 import * as React from 'react';
 
 export type LoadAssessmentButtonDeps = {
     detailsViewActionMessageCreator: DetailsViewActionMessageCreator;
     assessmentDataParser: AssessmentDataParser;
+    urlParser: UrlParser;
     loadAssessmentHelper: LoadAssessmentHelper;
 };
 export interface LoadAssessmentButtonProps {
     deps: LoadAssessmentButtonDeps;
+    tabStoreData: TabStoreData;
+    assessmentStoreData: AssessmentStoreData;
+}
+export interface LoadAssessmentButtonState {
+    newTargetPageData: Tab;
+    prevTargetPageData: Tab;
+    loadedAssessmentData?: VersionedAssessmentData;
+    show: boolean;
 }
 
 export const loadAssessmentButtonAutomationId = 'load-assessment-button';
 
-export class LoadAssessmentButton extends React.Component<LoadAssessmentButtonProps> {
+export class LoadAssessmentButton extends React.Component<
+    LoadAssessmentButtonProps,
+    LoadAssessmentButtonState
+> {
+    public constructor(props) {
+        super(props);
+        this.state = {
+            loadedAssessmentData: null,
+            newTargetPageData: this.getNewTab(),
+            prevTargetPageData: props.assessmentStoreData.persistedTabInfo,
+            show: false,
+        };
+    }
+
+    private toggleLoadDialog = () => {
+        this.setState(prevState => ({ show: !prevState.show }));
+    };
+
+    private getNewTab(): Tab {
```

Comment:
Not sure if I follow the intention behind this code: this is only called in the constructor (and that's only called on creation of the component). As a result, it will always return the tab store data and we never enter the if-block. Is there something I'm missing here?

Reply:
this was needed but I was missing where it should've been called, I added the logic in that uses this as intended.